families/track_and_trade/client/src/views/list_fish.js

```diff
@@ -0,0 +1,147 @@
+/**
+ * Copyright 2017 Intel Corporation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ * ----------------------------------------------------------------------------
+ */
+'use strict'
+
+const m = require('mithril')
+const truncate = require('lodash/truncate')
+const moment = require('moment')
+const {Table, FilterGroup, PagingButtons} = require('../components/tables')
+const api = require('../services/api')
+
+const PAGE_SIZE = 50
+
+const FishList = {
+  oninit (vnode) {
+    vnode.state.records = []
+    vnode.state.filteredRecords = []
+
+    vnode.state.currentPage = 0
+
+    api.get('/records').then((records) => {
+      vnode.state.records = records
+      vnode.state.filteredRecords = records
```

Comment:
It's not redundant, since all the filtering is happening on the client side. We need to keep the original list so we can perform the filtering actions.  The initial value is unfiltered, so we set both values to records.